apps/site/lib/site_web/templates/layout/_alert_announcement.html.eex

```diff
@@ -1,3 +1,6 @@
+<% alert = Alerts.Repo.by_id(@alert_banner.id) %>
+<%= if alert do %>
 <div class="alert-announcement__header" onclick="location.href='<%= @alert_banner.url || alert_path(@conn, :index) %>';">
-  <%= SiteWeb.AlertView.render "_banner.html", alert: %{Alerts.Repo.by_id(@alert_banner.id) | header: @alert_banner.title, priority: :system} %>
+  <%= SiteWeb.AlertView.render "_banner.html", alert: %{alert | header: @alert_banner.title, priority: :system} %>
```

Comment:
Isn't `@alert_banner` already an alert? Why the extra fetch?

Reply:
`Alerts.Alert.t()` and `Alerts.Banner.t()` are _slightly_ different for reasons I don't completely understand yet, but a few notes in [`Alerts.Cache.Store`](https://github.com/mbta/dotcom/commit/eaac9fdb9b32f0c55f64689ec23de01de283a513) are here.. `alert_banner` doesn't seem used in many places beyond this though, so maybe I should try refactoring it out?

Reply:
The main difference is that what we want to show for the `header` in a banner alert is separate from the actual `header` that's returned for the alert, and that we're overriding the priority. We should be able to address that by either 1) (easer) giving `Banner.t()` a `header` instead of a `title` 2) creating a Protocol that both `Alert.t` and `Banner.t` can implement which provides the relevant data for both types.

Reply:
Looking at it more closely, the `_banner.html` template relies on certain fields that are **not** present in `Banner.t()` - `priority` to render the icon, and `lifecycle` to generate a CSS class name. I guess this is why the extra fetch. I'm intrigued by the Protocol idea but am not sure what it's solving there?

Reply:
It would allow you to pass a `Banner.t()` to that template, by providing an interface to get the appropriate priority/lifecycle. Adding those fields to the Banner struct works out about the same.